### DISCUSSION TITLE
Is there an easier way to add videos from Youtube, vimeo, etc?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#2352] [Q&A]
Right now do I use the `iframe` option of YouTube and co. to embed a video into the docs.
But it looks bad for me, so I was wondering, if there is a sort of markdown equivalent to utilize. Maybe similar to how you add buttons? But you would f.e. use `[video](https://youtube.com/watch?v=<something>){: .md-ytvideo }`?

**Comment by squidfunk** [comment]
If there's an extension that adds this functionality (i.e. converts youtube URLs to the respective HTML), we may consider adding support for it. Maybe something that could be added to [MagicLink](https://facelessuser.github.io/pymdown-extensions/extensions/magiclink/), if there isn't already an extension that provides this feature.

**Comment by hellt** [comment]
yeah, I'd love to have an (insiders) feature that takes in a regular YT link and transforms it to iframe equivalent

**Reply by squidfunk** [reply]
I think by now there are Markdown extensions that offer this behavior. I haven't used one yet, but maybe other users can share.

**Reply by facelessuser** [reply]
Originally, I had been working on an extension found in this branch: https://github.com/facelessuser/pymdown-extensions/pull/1465. I created it as it had been requested at some point. I had hoped to get real feedback on it, but sadly, never got any from the original requestor.  I guess it turned out to not be as desired as originally thought, or they got used to manually including the HTML.

Regardless, I was a little uncertain of the approach, so I had been hoping to get more feedback, but the general lack of interest had me put it on the back burner.

Fast forward to today, I have been working on general-purpose blocks which are inching closer to completion. I have potentially considered implementing such media links under general-purpose blocks once they are completed, but again, no decisions have been made.

Now, there may be other solutions already available, and if so, you can probably use those now, and maybe that means such a solution isn't directly needed by Pymdown Extensions. I have not re-evaluated the extension/plugin landscape for Python Markdown to see if such solutions already exist now.

**Comment by hellt** [comment]
If you want someone to provide feedback for the work you're doing - I will
be glad to do so. My main services that i embed frequently is YouTube and
twitter

**Comment by StefanoCecere** [comment]
i could jump to insiders if there was a `![](https://youtube.com/watch?v=<something>)` auto embed .. like Obisdian does!

**Reply by kamilkrzyskow** [reply]
The https://github.com/soulless-viewer/mkdocs-video plugin kind of allows for a similar syntax. Tough there won't be any updates, so perhaps an Insiders only markdown extension allowing for such syntax would be still worthwhile developing 😆
